### PR TITLE
CPLP-1599 delete appuser role

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1599

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 1]
```diff
@@ -305,4 +306,31 @@ public Offer CreateOffer(string provider, OfferTypeId offerType, Action<Offer>?
...
+                offer.UserRoles.Any(userRole => userRole.Id == roleId)
+            ))
+            .SingleOrDefaultAsync();
+
+    ///<inheritdoc/>
+    public async Task DeleteAppRoleAsync(Guid appId, Guid roleId)
+    {
+        try
+        {
+            var userRole = new UserRole(roleId, string.Empty, appId);
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:326]: only the primary key is required to remove the entity. The appId is not part of the primary key, only the id is.
Initialize the to be deleted entity as ```new UserRole(roleId,null!,null!);```

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 2]
```diff
@@ -305,4 +306,31 @@ public Offer CreateOffer(string provider, OfferTypeId offerType, Action<Offer>?
...
+            _context.UserRoles.Attach(userRole);
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:327]: calling attach before remove is redundant.

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 3]
```diff
@@ -305,4 +306,31 @@ public Offer CreateOffer(string provider, OfferTypeId offerType, Action<Offer>?
...
+            _context.UserRoles.Remove(userRole);
+            await _context.SaveChangesAsync().ConfigureAwait(false);
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:329]: for consistency with other code the call to SaveChangesAsync shall be done in the businesslogic.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: fixed

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**